core/src/test/java/io/questdb/griffin/AlterTableSetTypeRestartTest.java

```diff
@@ -0,0 +1,384 @@
+                drainWalQueue(engine);
+                assertTrue(engine.isWalTable(token));
+                assertNumOfRows(engine, tableName, 9);
+            }
+            validateShutdown();
+        });
+    }
+
+    @Test
+    public void testWalToNonWalWithDropTable() throws Exception {
+        TestUtils.assertMemoryLeak(() -> {
+            try (final ServerMain questdb = new ServerMain("-d", root.toString(), Bootstrap.SWITCH_USE_DEFAULT_LOG_FACTORY_CONFIGURATION)) {
+                questdb.start();
+                createTable(tableName, "WAL");
+
+                final CairoEngine engine = questdb.getCairoEngine();
+                final TableToken token = engine.getTableToken(tableName);
+
+                insertInto(tableName);
+                drainWalQueue(engine);
+
+                // WAL table
+                assertTrue(engine.isWalTable(token));
+                assertNumOfRows(engine, tableName, 1);
+                assertConvertFileDoesNotExist(engine, tableName);
+
+                // schedule table conversion to non-WAL
+                setType(tableName, "BYPASS WAL");
+                drainWalQueue(engine);
+                final Path path = assertConvertFileExists(engine, tableName);
+                assertConvertFileContent(path, NON_WAL);
+
+                insertInto(tableName);
+                drainWalQueue(engine);
+                assertTrue(engine.isWalTable(token));
+                assertNumOfRows(engine, tableName, 2);
+
+                // drop table
+                dropTable(tableName);
+                drainWalQueue(engine);
+            }
+            validateShutdown();
+
+            // restart
+            try (final ServerMain questdb = new ServerMain("-d", root.toString(), Bootstrap.SWITCH_USE_DEFAULT_LOG_FACTORY_CONFIGURATION)) {
+                questdb.start();
+
+                final CairoEngine engine = questdb.getCairoEngine();
+                try {
+                    engine.getTableToken(tableName);
+                } catch (CairoException e) {
+                    assertTrue(e.getMessage().contains("[-1] table does not exist [table=testtable]"));
+                }
+            }
+        });
+    }
+
+    private static void assertConvertFileContent(Path convertFilePath, byte expected) throws IOException {
+        final byte[] fileContent = java.nio.file.Files.readAllBytes(java.nio.file.Paths.get(convertFilePath.toString()));
+        assertEquals(1, fileContent.length);
+        assertEquals(expected, fileContent[0]);
+    }
+
+    private static void assertConvertFileDoesNotExist(CairoEngine engine, String tableName) {
+        doesConvertFileExist(engine, tableName, false);
+    }
+
+    private static Path assertConvertFileExists(CairoEngine engine, String tableName) {
+        return doesConvertFileExist(engine, tableName, true);
+    }
+
+    private static void assertNumOfRows(CairoEngine engine, String tableName, int count) throws SqlException {
```

Comment:
this fails

```java
    @Test
    public void testWalToNonWal() throws Exception {
        TestUtils.assertMemoryLeak(() -> {
            try (final ServerMain questdb = new ServerMain("-d", root.toString(), Bootstrap.SWITCH_USE_DEFAULT_LOG_FACTORY_CONFIGURATION)) {
                questdb.start();
                createTable(tableName, "WAL");

                final CairoEngine engine = questdb.getCairoEngine();
                final TableToken token = engine.getTableToken(tableName);

                insertInto(tableName);
                drainWalQueue(engine);

                // WAL table
                assertTrue(engine.isWalTable(token));
                assertNumOfRows(engine, tableName, 1);
                assertConvertFileDoesNotExist(engine, tableName);

                // schedule table conversion to non-WAL
                setType(tableName, "BYPASS WAL");
                drainWalQueue(engine);
                final Path path = assertConvertFileExists(engine, tableName);
                assertConvertFileContent(path, NON_WAL);

                insertInto(tableName);
                drainWalQueue(engine);
                assertTrue(engine.isWalTable(token));
                assertNumOfRows(engine, tableName, 2);
            }
            validateShutdown();

            // restart
            try (final ServerMain questdb = new ServerMain("-d", root.toString(), Bootstrap.SWITCH_USE_DEFAULT_LOG_FACTORY_CONFIGURATION)) {
                questdb.start();

                final CairoEngine engine = questdb.getCairoEngine();
                final TableToken token = engine.getTableToken(tableName);
                assertFalse(engine.isWalTable(token));
            }
        });
    }
```

Reply:
passed for me and on CI too.
what is the error?

Reply:
The last assertion failed

```
assertFalse(engine.isWalTable(token));`
```

because the table wouldn't covert from WAL to non-WAL. It's a valid test, did you include it, I cannot see it?

Reply:
this test seems to work.
maybe recent changes made it pass.

Reply:
i can see you added it too, so i will merge